server/src/concepts/offer/repositories/commands.js

```diff
@@ -21,3 +21,19 @@ export const deleteOffer = async (offerId) => {
 export const updateOffer = async (filter, data) => {
   return await Offer.updateOne(filter, data);
 };
+
+export const activateOfferWithId = async (id) => {
+  return await Offer.findByIdAndUpdate(id, { isActivate: true });
+};
+
+export const archiveOfferWithId = async (id, userId) => {
+  const existingOffer = await Offer.findById(id);
+  if (existingOffer.userId !== userId) {
```

Comment:
```suggestion
  if (existingOffer.userId.toString() !== userId) {
```

existingOffer.userId is typeof objectId and userId is typeof string, so we need to convert objectId to string. More description on discord channel